modules/core/src/main/scala/lucuma/odb/api/model/ObservationModel.scala

```diff
@@ -153,13 +158,17 @@ object ObservationModel extends ObservationOptics {
       (existence   .validateIsNotNull("existence"),
        status      .validateIsNotNull("status"),
        activeStatus.validateIsNotNull("active"),
+       scienceRequirements.traverse(_.editor),
        constraintSet.traverse(_.editor)
-      ).mapN { (e, s, a, c) =>
+      ).mapN { (e, s, a, r, c) =>
         for {
           _ <- ObservationModel.existence    := e
           _ <- ObservationModel.name         := name.toOptionOption
           _ <- ObservationModel.status       := s
           _ <- ObservationModel.activeStatus := a
+          _ <- State.modify[ObservationModel] { o =>
+            r.fold(o) { ed => ObservationModel.scienceRequirements.modify(ed.runS(_).value)(o) }
+          }
```

Comment:
I wrote the similar case for `constraintSet` below but it bothered me that it was a bit awkward.  I'm sure there is a better way to do this.  One small improvement though would be:

```scala
          _ <- r.fold(State.get[ObservationModel].void) { ed =>
            ObservationModel.scienceRequirements.mod_(ed.runS(_).value)
          }
```

You have to import `import lucuma.core.optics.state.all._` for that.